nodejs/dist/uws.js

```diff
@@ -1,98 +1,134 @@
-/* ws interface for Node.js developers */
+'use strict';
 
-const NativeServer = require('./uws_' + process.platform + '_' + process.versions.modules).Server;
 const EventEmitter = require('events');
-const util = require('util');
-
-function Socket(nativeSocket, server) {
-    EventEmitter.call(this);
-    var self = this;
-    this.nativeSocket = nativeSocket;
-    this.server = server;
-
-    this.send = function(message, options) {
-        if (self.nativeSocket !== null) {
-            var binary = false;
-            if (options !== undefined) {
-                binary = options.binary;
-            }
-            self.server.nativeServer.send(self.nativeSocket, message, binary);
-        } else {
-            /* ignore sends on closed sockets */
-        }
+
+const uws = require(`./uws_${process.platform}_${process.versions.modules}`);
+const NativeServer = uws.Server;
+
+class Socket extends EventEmitter {
+    /**
+     * Creates a Socket instance.
+     *
+     * @param {Object} nativeSocket Socket instance
+     * @param {Server} server Server instance
+     */
+    constructor(nativeSocket, server) {
+        super();
+        this.nativeSocket = nativeSocket;
+        this.server = server;
+    }
+
+    /**
+     * Sends a message.
+     *
+     * @param {String|Buffer} message The message to send
+     * @param {Object} options Send options
+     * @return {Boolean} `true` if the message has been sent, else `false`
```

Comment:
Appreciate the thought behind this boolean. But I would like to keep the door open to an indicator being returned (could be boolean) to expose to the user that buffers are being filled up. This would allow an easy Writable stream wrapper around a Socket, which could be insanely useful in certain use cases (something `ws` *cannot* provide with its API).